iot/api-client/manager tests fail - Connection lost, ALREADY_EXISTS: Resource already exists in project.

Found in #4087 and many others.

```
------------------------------------------------------------
- testing iot/api-client/manager
------------------------------------------------------------
[ERROR] Tests run: 21, Failures: 0, Errors: 17, Skipped: 4, Time elapsed: 451.364 s <<< FAILURE! - in com.example.cloud.iot.examples.ManagerIT
[ERROR] com.example.cloud.iot.examples.ManagerIT.testMqttDeviceState  Time elapsed: 445.836 s  <<< ERROR!
org.eclipse.paho.client.mqttv3.MqttException: Connection lost
Caused by: javax.net.ssl.SSLException: Connection reset
Caused by: java.net.SocketException: Connection reset

[ERROR] com.example.cloud.iot.examples.ManagerIT.testHttpDeviceConfig  Time elapsed: 0.367 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testMqttDeviceConfig  Time elapsed: 0.231 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testMqttDeviceEvents  Time elapsed: 0.338 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateGetDevice  Time elapsed: 0.481 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testPatchEs  Time elapsed: 0.389 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateConfigureDevice  Time elapsed: 0.317 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateGetRegistry  Time elapsed: 0.346 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateListDevices  Time elapsed: 0.204 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateDeleteEsDevice  Time elapsed: 0.304 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testHttpDeviceEvent  Time elapsed: 0.316 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testHttpDeviceState  Time elapsed: 0.296 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testGetIam  Time elapsed: 0.47 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateDeleteRsaDevice  Time elapsed: 0.202 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testCreateDeleteUnauthDevice  Time elapsed: 0.294 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testSetIam  Time elapsed: 0.301 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] com.example.cloud.iot.examples.ManagerIT.testPatchRsa  Time elapsed: 0.403 s  <<< ERROR!
com.google.api.gax.rpc.AlreadyExistsException: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).
Caused by: io.grpc.StatusRuntimeException: ALREADY_EXISTS: Resource already exists in the project (resource=java-pst-16035045924).

[ERROR] Errors:
[ERROR]   ManagerIT.testCreateConfigureDevice » AlreadyExists io.grpc.StatusRuntimeExcep...
[ERROR]   ManagerIT.testCreateDeleteEsDevice » AlreadyExists io.grpc.StatusRuntimeExcept...
[ERROR]   ManagerIT.testCreateDeleteRsaDevice » AlreadyExists io.grpc.StatusRuntimeExcep...
[ERROR]   ManagerIT.testCreateDeleteUnauthDevice » AlreadyExists io.grpc.StatusRuntimeEx...
[ERROR]   ManagerIT.testCreateGetDevice » AlreadyExists io.grpc.StatusRuntimeException: ...
[ERROR]   ManagerIT.testCreateGetRegistry » AlreadyExists io.grpc.StatusRuntimeException...
[ERROR]   ManagerIT.testCreateListDevices » AlreadyExists io.grpc.StatusRuntimeException...
[ERROR]   ManagerIT.testGetIam » AlreadyExists io.grpc.StatusRuntimeException: ALREADY_E...
[ERROR]   ManagerIT.testHttpDeviceConfig » AlreadyExists io.grpc.StatusRuntimeException:...
[ERROR]   ManagerIT.testHttpDeviceEvent » AlreadyExists io.grpc.StatusRuntimeException: ...
[ERROR]   ManagerIT.testHttpDeviceState » AlreadyExists io.grpc.StatusRuntimeException: ...
[ERROR]   ManagerIT.testMqttDeviceConfig » AlreadyExists io.grpc.StatusRuntimeException:...
[ERROR]   ManagerIT.testMqttDeviceEvents » AlreadyExists io.grpc.StatusRuntimeException:...
[ERROR]   ManagerIT.testMqttDeviceState » Mqtt Connection lost
[ERROR]   ManagerIT.testPatchEs » AlreadyExists io.grpc.StatusRuntimeException: ALREADY_...
[ERROR]   ManagerIT.testPatchRsa » AlreadyExists io.grpc.StatusRuntimeException: ALREADY...
[ERROR]   ManagerIT.testSetIam » AlreadyExists io.grpc.StatusRuntimeException: ALREADY_E...
[ERROR] Tests run: 21, Failures: 0, Errors: 17, Skipped: 4
```

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -46,10 +46,6 @@ import org.junit.runners.JUnit4;
 @RunWith(JUnit4.class)
 @SuppressWarnings("checkstyle:abbreviationaswordinname")
 public class ManagerIT {
-  private ByteArrayOutputStream bout;
-  private PrintStream out;
-  private DeviceRegistryExample app;
-
   private static final String CLOUD_REGION = "us-central1";
   private static final String ES_PATH = "resources/ec_public.pem";
   private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
@@ -64,9 +60,11 @@ public class ManagerIT {
           + UUID.randomUUID().toString().substring(0, 20);
   private static final String MEMBER = "group:dpebot@google.com";
   private static final String ROLE = "roles/viewer";
-
   private static Topic topic;
   private static boolean hasCleared = false;
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+  private DeviceRegistryExample app;
 
   @Before
   public void setUp() throws Exception {

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -28,13 +28,11 @@ import com.google.auth.oauth2.GoogleCredentials;
 import com.google.cloud.pubsub.v1.TopicAdminClient;
 import com.google.pubsub.v1.ProjectTopicName;
 import com.google.pubsub.v1.Topic;
-
 import java.io.ByteArrayOutputStream;
 import java.io.PrintStream;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
 import java.util.UUID;
-
 import org.eclipse.paho.client.mqttv3.MqttClient;
 import org.junit.After;
 import org.junit.Assert;

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -28,10 +28,13 @@ import com.google.auth.oauth2.GoogleCredentials;
 import com.google.cloud.pubsub.v1.TopicAdminClient;
 import com.google.pubsub.v1.ProjectTopicName;
 import com.google.pubsub.v1.Topic;
+
 import java.io.ByteArrayOutputStream;
 import java.io.PrintStream;
 import java.nio.charset.StandardCharsets;
 import java.util.List;
+import java.util.UUID;
+
 import org.eclipse.paho.client.mqttv3.MqttClient;
 import org.junit.After;
 import org.junit.Assert;
@@ -52,10 +55,18 @@ public class ManagerIT {
   private static final String CLOUD_REGION = "us-central1";
   private static final String ES_PATH = "resources/ec_public.pem";
   private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
-  private static final String REGISTRY_ID = "java-reg-" + (System.currentTimeMillis() / 100L);
+  private static final String REGISTRY_ID =
+      "java-reg-"
+          + (System.currentTimeMillis() / 100L)
+          + "-"
+          + UUID.randomUUID().toString().substring(0, 10);
   private static final String RSA_PATH = "resources/rsa_cert.pem";
   private static final String PKCS_PATH = "resources/rsa_private_pkcs8";
-  private static final String TOPIC_ID = "java-pst-" + (System.currentTimeMillis() / 100L);
+  private static final String TOPIC_ID =
+      "java-pst-"
+          + (System.currentTimeMillis() / 100L)
+          + "-"
+          + UUID.randomUUID().toString().substring(0, 10);
   private static final String MEMBER = "group:dpebot@google.com";
   private static final String ROLE = "roles/viewer";
 

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -55,6 +55,7 @@ public class ManagerIT {
   private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
   private static final String REGISTRY_ID =
       "java-reg-"
+              + UUID.randomUUID().toString().substring(0, 10) + "-"
           + (System.currentTimeMillis() / 100L);
   private static final String RSA_PATH = "resources/rsa_cert.pem";
   private static final String PKCS_PATH = "resources/rsa_private_pkcs8";
@@ -110,7 +111,7 @@ public class ManagerIT {
         if (registryId.startsWith("java-reg-")) {
           long currSecs = System.currentTimeMillis() / 1000L;
           long regSecs =
-              Long.parseLong(registryId.substring("java-reg-".length(), registryId.length()));
+              Long.parseLong(registryId.substring("java-reg-".length() + 11, registryId.length()));
           long diffSecs = currSecs - regSecs;
           if (diffSecs > (60 * 60 * 24 * 7 * 10)) { // tests from last week or older
             System.out.println("Remove Id: " + r.getId());

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -55,9 +55,7 @@ public class ManagerIT {
   private static final String PROJECT_ID = System.getenv("GOOGLE_CLOUD_PROJECT");
   private static final String REGISTRY_ID =
       "java-reg-"
-          + (System.currentTimeMillis() / 100L)
-          + "-"
-          + UUID.randomUUID().toString().substring(0, 10);
+          + (System.currentTimeMillis() / 100L);
   private static final String RSA_PATH = "resources/rsa_cert.pem";
   private static final String PKCS_PATH = "resources/rsa_private_pkcs8";
   private static final String TOPIC_ID =

--- a/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
+++ b/iot/api-client/manager/src/test/java/com/example/cloud/iot/examples/ManagerIT.java
@@ -60,9 +60,7 @@ public class ManagerIT {
   private static final String PKCS_PATH = "resources/rsa_private_pkcs8";
   private static final String TOPIC_ID =
       "java-pst-"
-          + (System.currentTimeMillis() / 100L)
-          + "-"
-          + UUID.randomUUID().toString().substring(0, 10);
+          + UUID.randomUUID().toString().substring(0, 20);
   private static final String MEMBER = "group:dpebot@google.com";
   private static final String ROLE = "roles/viewer";
 

